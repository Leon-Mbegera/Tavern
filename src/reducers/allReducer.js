import {
  GET_COCKTAILS_REQUEST, GET_COCKTAILS_SUCCESS,
  GET_COCKTAILS_FAILURE,
} from '../actions/index';

const initialState = {
  cocktails: [],
  error: '',
};

const allReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_COCKTAILS_REQUEST:
      return {
        ...state,
        error: '',
      };
    case GET_COCKTAILS_SUCCESS:
      return {
        ...state,
        cocktails: action.payload,
      };
    case GET_COCKTAILS_FAILURE:
      return {
        ...state,
        error: action.payload,
      };
    default:
      return state;
  }
};

export default allReducer;
