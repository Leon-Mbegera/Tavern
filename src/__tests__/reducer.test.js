import allReducer from '../reducers/allReducer';
import {
  GET_COCKTAILS_REQUEST, GET_COCKTAILS_SUCCESS,
} from '../actions/index';

const initialState = {
  cocktails: [],
  error: '',
};

describe('All Cocktails Reducer', () => {
  test('should return the initial state', () => {
    expect(allReducer(undefined, {})).toEqual({
      cocktails: [],
      error: '',
    });
  });

  it('returned initial state should match snapshot', () => {
    expect(allReducer(undefined, {})).toMatchSnapshot({
      cocktails: [],
      error: '',
    });
  });

  test('should handle GET_COCKTAILS_REQUEST', () => {
    expect(allReducer(initialState, GET_COCKTAILS_REQUEST)).toEqual({
      ...initialState,
      error: '',
    });
  });

  test('should handle GET_COCKTAIL_SUCCESS', () => {
    const response = {
      cocktails: [
        { strDrink: '155 Belmont' },
      ],
    };

    const successAction = {
      type: GET_COCKTAILS_SUCCESS,
      payload: response.cocktails,
    };

    expect(allReducer(initialState, successAction)).toEqual({
      ...initialState,
      cocktails: successAction.payload,
    });
  });

  test('should handle GET_COCKTAILS_FAILURE', () => {
    const response = {
      error: "404 Error couldn't fetch",
    };

    const failureAction = {
      type: GET_COCKTAILS_SUCCESS,
      payload: response.error,
    };

    expect(allReducer(initialState, failureAction)).toEqual({
      ...initialState,
      cocktails: failureAction.payload,
    });
  });
});
