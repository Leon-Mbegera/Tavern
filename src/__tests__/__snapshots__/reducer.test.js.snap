// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Cocktails Reducer returned initial state should match snapshot 1`] = `
Object {
  "cocktails": Array [],
  "error": "",
}
`;
